File: appindexes/app/src/main/java/com/xuebinduan/app/MainActivity.java
package com.xuebinduan.app;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.os.Bundle;

import com.negier.choosephoto.ChoosePhotoMainActivity;
import com.negier.contactlistview.ContactsActivity;
import com.negier.coordinatorlayoutdemo.CoordinatorLayoutMainActivity;
import com.negier.coverbaidulangbottom.CoverBaiduLangBottomMainActivity;
import com.negier.loop.bestinfiniteloop.BestInfiniteLoopActivity;
import com.xuebinduan.app.databinding.ActivityMainBinding;
import com.xuebinduan.base64.Base64Activity;
import com.xuebinduan.keyboardimeoptions.IMEOptionsActivity;
import com.xuebinduan.looknewaddfile2.SplashLookNewAddFileActivity;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar!=null){
            actionBar.setTitle("主页");
        }

        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        StaggeredGridLayoutManager staggeredGridLayoutManager = new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL);
        mBinding.recyclerView.setLayoutManager(staggeredGridLayoutManager);
        ItemBean[] itemBeans = loadData();
        mBinding.recyclerView.setAdapter(new RecyclerViewAdapter(itemBeans));
    }

    private ItemBean[] loadData() {
        ItemBean[] itemBeans = new ItemBean[]{
                new ItemBean(R.mipmap.keyboardimeoptions,getString(R.string.keyboardimeoptions), IMEOptionsActivity.class),
                new ItemBean(R.mipmap.base64,getString(R.string.base64), Base64Activity.class),
                new ItemBean(R.mipmap.looknewaddfile,getString(R.string.looknewaddfile), SplashLookNewAddFileActivity.class),
                new ItemBean(R.mipmap.bestinfiniteloop,getString(R.string.bestinfiniteloop), BestInfiniteLoopActivity.class),
                new ItemBean(R.mipmap.choosephoto,getString(R.string.choosephoto), ChoosePhotoMainActivity.class),
                new ItemBean(R.mipmap.contactlist,getString(R.string.contactlist), ContactsActivity.class),
                new ItemBean(R.mipmap.img_black,getString(R.string.coordinatorlayout), CoordinatorLayoutMainActivity.class),
                new ItemBean(R.mipmap.coverbaidulangbottom,getString(R.string.coverbaidulangbottom), CoverBaiduLangBottomMainActivity.class)
        };
        return itemBeans;
    }

}

File: study_abroad/HenCoder/CustomViewNestedScroll/src/main/java/com/xuebinduan/customviewnestedscroll/MainActivity.kt
package com.xuebinduan.customviewnestedscroll

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

//https://www.jianshu.com/p/e333f11f29aa
//https://github.com/szn0212/NestedScrollingExample/tree/cecc194a0a0407793c9a44b6efeed997e789d6bc\

/**
 * NestedScrolling机制：
 * 当子view开始滚动之前，可以通知父view，让其先于自己进行滚动;
 * 子view滚动之后，还可以通知 父view 继续滚动。
 * 在这套交互机制中，child 是动作的发起者，parent 只是接受回调并作出响应。
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_nest)



    }
}